cognite/client/data_classes/simulators/routine_revisions.py

```diff
@@ -0,0 +1,653 @@
+        order (int): The order of the stage.
+        steps (list[SimulatorRoutineStep]): The steps of the stage.
+        description (str | None): The description of the stage.
+    """
+
+    order: int
+    steps: list[SimulatorRoutineStep]
+    description: str | None
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            order=resource["order"],
+            steps=[
+                SimulatorRoutineStep._load(step_, cognite_client) if isinstance(step_, dict) else step_
+                for step_ in resource["steps"]
+            ],
+            description=resource.get("description"),
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case=camel_case)
+        output["steps"] = [step_.dump(camel_case=camel_case) for step_ in self.steps]
+        return output
+
+
+class SimulatorRoutineRevisionCore(WriteableCogniteResource["SimulatorRoutineRevisionWrite"], ABC):
+    def __init__(
+        self,
+        external_id: str,
+        routine_external_id: str,
+        configuration: SimulatorRoutineConfiguration | None = None,
+        script: list[SimulatorRoutineStage] | None = None,
+    ) -> None:
+        self.external_id = external_id
+        self.routine_external_id = routine_external_id
+        self.configuration = configuration
+        self.script = script
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case=camel_case)
+        output["configuration"] = self.configuration.dump(camel_case=camel_case) if self.configuration else None
+        output["script"] = [stage_.dump(camel_case=camel_case) for stage_ in self.script] if self.script else None
+
+        return output
+
+
+class SimulatorRoutineRevisionWrite(SimulatorRoutineRevisionCore):
+    """
+    The simulator routine resource defines instructions on interacting with a simulator model.
+    This is a writeable version of a simulator routine revision, it is used when creating simulator routine revisions.
+
+    Args:
+        external_id (str): The external ID provided by the client. Must be unique for the resource type.
+        routine_external_id (str): The external ID of the simulator routine.
+        configuration (SimulatorRoutineConfiguration | None): The configuration of the simulator routine revision.
+        script (list[SimulatorRoutineStage] | None): The script of the simulator routine revision.
+
+    """
+
+    def __init__(
+        self,
+        external_id: str,
+        routine_external_id: str,
+        configuration: SimulatorRoutineConfiguration | None = None,
+        script: list[SimulatorRoutineStage] | None = None,
+    ) -> None:
+        super().__init__(
+            external_id=external_id,
+            routine_external_id=routine_external_id,
+            configuration=configuration,
+            script=script,
+        )
```

Comment:
Is it necessary to repeat the method? You already have it form the parent classi
```suggestion
```